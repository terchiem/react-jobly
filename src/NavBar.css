.NavBar {
  display: flex;
  justify-content: space-around;
  padding: 20px;
  margin-bottom: 20px;
  background-color: #fff;
  box-shadow: 1px 1px 1px 1px rgba(0, 0, 0, 0.1);
}

.NavBar a {
  text-decoration: none;
  color: #2469ff;
}

.NavBar-left {
  justify-self: flex-start;
}

.NavBar .active {
  font-weight: 900;
  color: #222;
}