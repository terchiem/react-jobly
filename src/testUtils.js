import React from 'react';
import UserContext from './UserContext';

const testUser = {
  username: "test",
  first_name: "test",
  last_name: "test",
  email: "test@email.com",
  photo_url: null
}

const testToken = "test";

function UserProvider({ children, currentUser=testUser, token=testToken }) {
  return (
    <UserContext.Provider value={{currentUser, token}}>
      {children}
    </UserContext.Provider>
  )
};

export { UserProvider };