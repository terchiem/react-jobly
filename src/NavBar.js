import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import './NavBar.css';

/** 
 * Navigation component for the Jobly app. 
 * Changes depending if user is logged in. 
 * */

function NavBar() {

  return (
    <nav className='NavBar'>
      <Link to='/' className='NavBar-left'>Jobly</Link>

      <NavLink exact to='/companies'>Companies</NavLink>
      <NavLink exact to='/jobs'>Jobs</NavLink>
      <NavLink exact to='/login'>Login</NavLink>
      <NavLink exact to='/profile'>Profile</NavLink>
    </nav>
  )
}

export default NavBar;