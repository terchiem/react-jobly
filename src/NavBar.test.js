import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import NavBar from './NavBar';
import { UserProvider } from './testUtils';

test('renders without crashing', () => {
  render(
    <MemoryRouter>
      <UserProvider>
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );
});

test('matches snapshot', () =>{
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('matches snapshot when logged out', () =>{
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider token={null} >
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});