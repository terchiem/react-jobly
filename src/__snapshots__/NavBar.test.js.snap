// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBar"
  >
    <a
      href="/"
      id="NavBar-logo"
    >
      Jobly
    </a>
    <ul>
      <li>
        <a
          href="/companies"
        >
          Companies
        </a>
      </li>
      <li>
        <a
          href="/jobs"
        >
          Jobs
        </a>
      </li>
      <li>
        <a
          href="/profile"
        >
          Profile
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Log Out
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="NavBar"
  >
    <a
      href="/"
      id="NavBar-logo"
    >
      Jobly
    </a>
    <ul>
      <li>
        <a
          href="/login"
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
